prow/spyglass/lenses/junit/lens.go

```diff
@@ -120,13 +148,27 @@ func (lens Lens) Body(artifacts []lenses.Artifact, resourceDir string, data stri
 				for _, subSuite := range suite.Suites {
 					record(subSuite)
 				}
+
 				for _, test := range suite.Results {
-					result.junit = append(result.junit, test)
+					test := test
+					// There are cases where multiple entries of exactly the same
+					// testcase in a single junit result file, this could result
+					// from reruns of test cases by `go test --count=N` where N>1.
+					// Deduplicate them here in this case, and classify a test as being
+					// flaky if it both succeeded and failed
+					testFullName := fmt.Sprintf("%s...%s...%s", suite.Name, test.ClassName, test.Name)
```

Comment:
```golang
type foo struct {
  file string
  class string
  test string
}
```

Seems like a more appropriate key than string mangling.

Reply:
great to learn. done